spec/The-OpenFF-System.md

```diff
@@ -0,0 +1,185 @@
+independent_variables={"x"}) potential_handler.expression
+```
+
+| MUST | MAY |  MUST NOT |
+|---|---|---|
+| Encode an energy evaluation method | Encode either an algebraic expression or tabluar data |
+| Also encode the necessary data to evaluate the functional form |
+| Import from some common base class
+| Include methods to expose parameter sources
+| Include methods to expose differentiable representations of data
+
+### Chemical topology
+
+The [OpenFF Topology](https://open-forcefield-toolkit.readthedocs.io/en/0.7.2/api/generated/openforcefield.topology.Topology.html#openforcefield.topology.Topology) (or some future form of it) will serve as the representation of the chemical topology. This encodes a graph-like representation of the system _but no physics_. The envisioned scope of this component is roughly equivalent to the data that can be stored in an SDF file.
+
+| MUST | MAY |  MUST NOT |
+|---|---|---|
+| | | |
+| Atoms with elements | Metadata of cheminformatics methods | Force field data |
+| Bonds | Bond constraints
+| | Fractional bond orders
+| Iterators for other valences | Residue/chain/segments
+| Periodicity boolean | Box vectors
+| | | |
+
+### Positions
+
+Positions of all particles are stored in a unit-bearing Nx3 matrix. Here, particles refers to all atoms and virtual sites in the system.
+
+| MUST | MAY |  MUST NOT |
+|---|---|---|
+| Unit-tagged Nx3 array of atomic positions | Unit-tagged virtual site potitions | Positions with implicit or ambiguous units
+
+### Box vectors
+
+Box vectors are represented as a 3x3 matrix of unit-bearing quantities that specify the periodicity of the dimensions of a periodic simulation cell. Both orthogonal and triclinic boxes are supported. For details, see the [OpenMM implementation](http://docs.openmm.org/latest/userguide/theory.html#periodic-boundary-conditions).
+
+Box vectors are optional. In order to represent non-periodic (i.e. vacuum) systems, `None` is an valid value.
+
+| MUST | MAY |  MUST NOT |
+|---|---|---|
+| Unit-tagged 3x3 array of atomic positions | `None` to encode non-periodicity | Positions with implicit or ambiguous units
+
+## Interoperability
+
+In contrast to existing efforts, this specification does not assume first-class compatibility with any particular molecular simulation engines or software stacks. It instead aims to provide a general and flexible model for storing data and separate out interoperability tasks as a separate layer.
+
+Separating the internal representation from the interoperability layer allows for a clear layer between them in which sanity checks can be carried out and compand compatibility checks can be encoded. For example, if an object in memory includes CMAP terms, it cannot be written to Cassandra, which does not support CMAPs, or if an object in memory includes Buckingham potentials, it cannot be converted to a ParmEd Structure.
+
+## Features
+
+### Tracking parameter sources
```

Comment:
Can you say a little more about what the data structures are that store/represent provenance?